src/PaperG/FirehoundBlob/Facebook/FacebookCreative.php

```diff
@@ -105,18 +126,26 @@ public function toArray()
      */
     public function fromArray($array)
     {
-        $this->type = $this->safeGet($array, self::TYPE);
-        $objects = $this->safeGet($array, self::OBJECTS, []);
-        $objectArray = [];
-        foreach ($objects as $object) {
-            if ($this->type == 'carousel') {
-                $objectArray[] = new FacebookCarouselCreativeData($object);
-            } else {
-                $objectArray[] = new FacebookCreativeData($object); //versioned, might need builder
-            }
+        $this->type       = $this->safeGet($array, self::TYPE);
+        $primary    = $this->safeGet($array, self::PRIMARY);
+        $childAttachments = $this->safeGet($array, self::CHILD_ATTACHMENTS, []);
+
+        $this->primary = $this->createCreativeData($primary);
+
+        $objectsArray = [];
+        foreach ($childAttachments as $object) {
+            $objectsArray[]  = $this->createCreativeData($object);
         }
 
-        $this->objects = $objectArray;
+        $this->childAttachments = $objectsArray;
     }
 
+    private function createCreativeData($creative)
+    {
+        if ($this->type == self::CAROUSEL_TYPE) {
+            return new FacebookCarouselCreativeData($creative);
+        } else {
+            return new FacebookCreativeData($creative);
+        }
+    }
```

Comment:
Well you want to make sure that it's still interoperable in classes that use this class.  If you maintain that then the `extend `works.